### ISSUE TITLE
Certificate chain error on minikube windows

### ISSUE THREAD

**Issue by mgj** [kubernetes-client/csharp#600]
I'm experiencing a "Certificate chain"-error when trying to access the Kubernetes API using the C# Client library.

Minikube version: v1.18.1
C# Client library version: 4.0.21

I create a new Kubernetes instance, which succeeds: `_kubernetes = new Kubernetes(KubernetesClientConfiguration.InClusterConfig())`

I then try to use the kubernetes client: `await _kubernetes.ListNamespaceAsync()` but this causes an error:

### Kubernetes.ListnamespaceAsync stacktrace
```
The remote certificate is invalid because of errors in the certificate chain: PartialChain

at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
         at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)
         at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
```
```
The SSL connection could not be established, see inner exception.

at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
         at k8s.WatcherDelegatingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
         at k8s.Kubernetes.ListNamespaceWithHttpMessagesAsync(Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, String pretty, Dictionary`2 customHeaders, CancellationToken cancellationToken)
         at k8s.KubernetesExtensions.ListNamespaceAsync(IKubernetes operations, Nullable`1 allowWatchBookmarks, String continueParameter, String fieldSelector, String labelSelector, Nullable`1 limit, String resourceVersion, String resourceVersionMatch, Nullable`1 timeoutSeconds, Nullable`1 watch, String pretty, CancellationToken cancellationToken)
```

Why am seeing a certificate error? I am running this on minikube on windows with hyperv. I do have an nginx ingress in my cluster that does TLS termination, but I'm assuming this is not the certificate its referring to.

I have set up a kubernetes service account to enable access to the kubernetes API from my pod:

### Kubernetes ServiceAccount Helm Chart
```
kind: ServiceAccount
apiVersion: v1
metadata:
  name: my-serviceaccount
  namespace: {{ .Values.namespace }}
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: my-serviceaccount
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
  - kind: ServiceAccount
    name: my-serviceaccount
    namespace: {{ .Values.namespace }}
```

I have also attached my InCluster config encoded as json:

### InCluster config (json)
```
{
	"CurrentContext": null,
	"Namespace": null,
	"Host": "https://10.96.0.1:443/",
	"SslCaCerts": [{
		"Archived": false,
		"Extensions": [{
			"KeyUsages": 164,
			"Critical": true,
			"Oid": {
				"Value": "2.5.29.15",
				"FriendlyName": "X509v3 Key Usage"
			},
			"RawData": "AwICpA=="
		}, {
			"EnhancedKeyUsages": [{
				"Value": "1.3.6.1.5.5.7.3.2",
				"FriendlyName": "TLS Web Client Authentication"
			}, {
				"Value": "1.3.6.1.5.5.7.3.1",
				"FriendlyName": "TLS Web Server Authentication"
			}],
			"Critical": false,
			"Oid": {
				"Value": "2.5.29.37",
				"FriendlyName": "X509v3 Extended Key Usage"
			},
			"RawData": "MBQGCCsGAQUFBwMCBggrBgEFBQcDAQ=="
		}, {
			"CertificateAuthority": true,
			"HasPathLengthConstraint": false,
			"PathLengthConstraint": 0,
			"Critical": true,
			"Oid": {
				"Value": "2.5.29.19",
				"FriendlyName": "X509v3 Basic Constraints"
			},
			"RawData": "MAMBAf8="
		}, {
			"SubjectKeyIdentifier": "F2B071E886EBB26B7F1F4166A0CE52CEDEC006EE",
			"Critical": false,
			"Oid": {
				"Value": "2.5.29.14",
				"FriendlyName": "X509v3 Subject Key Identifier"
			},
			"RawData": "BBTysHHohuuya38fQWagzlLO3sAG7g=="
		}],
		"FriendlyName": "",
		"HasPrivateKey": false,
		"PrivateKey": null,
		"IssuerName": {
			"Name": "CN=minikubeCA",
			"Oid": {
				"Value": null,
				"FriendlyName": null
			},
			"RawData": "MBUxEzARBgNVBAMTCm1pbmlrdWJlQ0E="
		},
		"NotAfter": "2031-01-05T20:14:08+00:00",
		"NotBefore": "2021-01-06T20:14:08+00:00",
		"PublicKey": {
			"EncodedKeyValue": {
				"Oid": {
					"Value": "1.2.840.113549.1.1.1",
					"FriendlyName": "RSA"
				},
				"RawData": "MIIBCgKCAQEApuvAOVHV+h1ZT6OvczYt/1NDcctMMw2lDfYV/sRXeAjMbul/2x45IP7YfMfL8v2VNfqeLtdLTaXqsqMImAfF2LDrd4M/H0FtXeCnAmbtX87dzKkX2j7KwdekZpgB4OJKw13NlKhYDqEglHco5yweB7y/dIAyKoASWwZY7CoyJ8oX6qGfcUysc0d92d9pPWMzvMVXd4ug8UWOkY3+lKyzAmQUwTB6eWZ2pfju6yzS12y7sPZK/v4unayGnI2/3s3cqDHDsMIj0Mul9yxRy/Dq8NC9anzdIyUJkQ0B6+WqL6hjLHx3OZw2nXseKhL+ZNZRnpDEBje0pI8Mm2T+xs8O5QIDAQAB"
			},
			"EncodedParameters": {
				"Oid": {
					"Value": "1.2.840.113549.1.1.1",
					"FriendlyName": "RSA"
				},
				"RawData": "BQA="
			},
			"Key": {
				"LegalKeySizes": [{
					"MinSize": 512,
					"MaxSize": 16384,
					"SkipSize": 8
				}],
				"KeyExchangeAlgorithm": "RSA",
				"SignatureAlgorithm": "RSA"
			},
			"Oid": {
				"Value": "1.2.840.113549.1.1.1",
				"FriendlyName": "RSA"
			}
		},
		"RawData": "MIIDBjCCAe6gAwIBAgIBATANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwptaW5pa3ViZUNBMB4XDTIxMDEwNjIwMTQwOFoXDTMxMDEwNTIwMTQwOFowFTETMBEGA1UEAxMKbWluaWt1YmVDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKbrwDlR1fodWU+jr3M2Lf9TQ3HLTDMNpQ32Ff7EV3gIzG7pf9seOSD+2HzHy/L9lTX6ni7XS02l6rKjCJgHxdiw63eDPx9BbV3gpwJm7V/O3cypF9o+ysHXpGaYAeDiSsNdzZSoWA6hIJR3KOcsHge8v3SAMiqAElsGWOwqMifKF+qhn3FMrHNHfdnfaT1jM7zFV3eLoPFFjpGN/pSsswJkFMEwenlmdqX47uss0tdsu7D2Sv7+Lp2shpyNv97N3Kgxw7DCI9DLpfcsUcvw6vDQvWp83SMlCZENAevlqi+oYyx8dzmcNp17HioS/mTWUZ6QxAY3tKSPDJtk/sbPDuUCAwEAAaNhMF8wDgYDVR0PAQH/BAQDAgKkMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBTysHHohuuya38fQWagzlLO3sAG7jANBgkqhkiG9w0BAQsFAAOCAQEAW14u7qjsMbM/LXNKJ31j12KVAe/KpyWTvYvbppqv9ndWiGKIrM4CFjY+Tkw2zXQSSPSWGIORPPX8dTeIhqDRqNy4UyOIu4PBX/tQyacckbdiZ4dP8g0IScxLYMhPz99E9VuyK5MeiOJTzAvRYhc4530RXJjQxPBTzChmCbIudZls5U3HcYLclPDnta2RZknpc5KyYpxM/K8VCWr6G6kiZBwmOXvB9+NIPIPDjYrIXNFPrIuqF0xB6RR1VxIMAhiLz4pa1fK6h2UKCOPJDw5QdXu5TIG5do8/Y1D078TRzm+/hXzLRqGMQIaSDdNgWB40ZSFr3H8DcXYSSSNXNNedWw==",
		"SerialNumber": "01",
		"SignatureAlgorithm": {
			"Value": "1.2.840.113549.1.1.11",
			"FriendlyName": "sha256RSA"
		},
		"SubjectName": {
			"Name": "CN=minikubeCA",
			"Oid": {
				"Value": null,
				"FriendlyName": null
			},
			"RawData": "MBUxEzARBgNVBAMTCm1pbmlrdWJlQ0E="
		},
		"Thumbprint": "3D825CA8A09A24561EE16A9AF0609F697443351B",
		"Version": 3,
		"Handle": {
			"value": 140181022244400
		},
		"Issuer": "CN=minikubeCA",
		"Subject": "CN=minikubeCA"
	}],
	"ClientCertificateData": null,
	"ClientCertificateKeyData": null,
	"ClientCertificateFilePath": null,
	"ClientCertificateKeyStoreFlags": null,
	"ClientKeyFilePath": null,
	"SkipTlsVerify": false,
	"UserAgent": null,
	"Username": null,
	"Password": null,
	"AccessToken": null,
	"TokenProvider": {},
	"TcpKeepAlive": true
}
```

Crossposted: https://stackoverflow.com/questions/66935759/kubernetes-api-certificate-chain-error-on-minikube-windows

**Comment by tg123** [comment]
fixed in #585

please update to 4.0.22

**Comment by mgj** [comment]
Thank you for your quick response

4.0.22 works for me(!) Very cool, thank you
